applications/MeshingApplication/custom_utilities/mmg_utilities.cpp

```diff
@@ -0,0 +1,2556 @@
+    if (TMMGLibrary == MMGLibrary::MMG2D) { // 2D
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfLines << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTriangles << std::endl;
+    } else if (TMMGLibrary == MMGLibrary::MMG3D) { // 3D
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfTriangles + rMMGMeshInfo.NumberOfQuadrilaterals << "\n\tTriangles: " << rMMGMeshInfo.NumberOfTriangles << "\tQuadrilaterals: " << rMMGMeshInfo.NumberOfQuadrilaterals << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTetrahedra + rMMGMeshInfo.NumberOfPrism << "\n\tTetrahedron: " << rMMGMeshInfo.NumberOfTetrahedra << "\tPrisms: " << rMMGMeshInfo.NumberOfPrism << std::endl;
+    } else { // Surfaces
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfLines << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTriangles << std::endl;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<MMGLibrary TMMGLibrary>
+IndexVectorType MmgUtilities<TMMGLibrary>::CheckNodes(
+    ModelPart& rModelPart,
+    const IndexType EchoLevel
+    )
+{
+    DoubleVectorMapType node_map;
+
+    IndexVectorType nodes_to_remove_ids;
+
+    DoubleVectorType coords(Dimension);
+
+    auto& r_nodes_array = rModelPart.Nodes();
+    const auto it_node_begin = r_nodes_array.begin();
+
+    for(SizeType i = 0; i < r_nodes_array.size(); ++i) {
+        auto it_node = it_node_begin + i;
+
+        const array_1d<double, 3>& r_coordinates = it_node->Coordinates();
+
+        for(IndexType i_coord = 0; i_coord < Dimension; i_coord++)
+            coords[i_coord] = r_coordinates[i_coord];
+
+        node_map[coords] += 1;
+
+        if (node_map[coords] > 1) {
+            nodes_to_remove_ids.push_back(it_node->Id());
+            KRATOS_WARNING_IF("MmgUtilities", EchoLevel > 0) << "The mode " << it_node->Id() <<  " is repeated"<< std::endl;
+        }
+    }
+
+    return nodes_to_remove_ids;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+IndexVectorType MmgUtilities<MMGLibrary::MMG2D>::CheckFirstTypeConditions(const IndexType EchoLevel)
+{
+    IndexVectorMapType edge_map;
+
+    IndexVectorType ids(2);
+
+    IndexVectorType conditions_to_remove;
+
+    // Iterate in the conditions
+    for(int i = 0; i < mMmgMesh->na; ++i) {
+        int edge_0, edge_1, prop_id, is_ridge, is_required;
+
+        if (MMG2D_Get_edge(mMmgMesh, &edge_0, &edge_1, &prop_id, &is_ridge, &is_required) != 1 )
+            exit(EXIT_FAILURE);
```

Comment:
Wouldn't it make sense to throw an error here instead. I think it would give better debugging info.

Reply:
I use the API of MMG

Reply:
I'm not following. How does that prevent you from throwing?

Reply:
If we use our debug interface, we will lose the debug info from the library, which may help to fix problems

Reply:
ok, I just thought it would be helpful to identify the location in kratos where it fails.